cmd/cmd.go

```diff
@@ -34,10 +33,47 @@ import (
 	"github.com/uber-common/bark"
 	"github.com/uber/aresdb/clients"
 	"github.com/uber/aresdb/memutils"
+	"github.com/spf13/cobra"
 )
 
-// StartService is the entry point of starting ares.
-func StartService(cfg common.AresServerConfig, logger bark.Logger, queryLogger bark.Logger, metricsCfg common.Metrics, httpWrappers ...utils.HTTPHandlerWrapper) {
+// AppParams contains application parameters for creating command
+type AppParams struct {
+	DefaultCfg   map[string]interface{}
+	ServerLogger bark.Logger
+	QueryLogger bark.Logger
+	Metrics common.Metrics
```

Comment:
instead of using a big struct, probably we can use optional argument pattern here:
https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md
so user can specify what they want and leave defaults for remaining

Reply:
Done